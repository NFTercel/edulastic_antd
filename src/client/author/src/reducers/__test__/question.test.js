import questionReducer from "../question";

import {
  RECEIVE_QUESTION_REQUEST,
  RECEIVE_QUESTION_SUCCESS,
  RECEIVE_QUESTION_ERROR,
  SAVE_QUESTION_REQUEST,
  SAVE_QUESTION_SUCCESS,
  SAVE_QUESTION_ERROR,
  SET_QUESTION_DATA,
  SET_QUESTION,
  SET_QUESTION_ALIGNMENT_ADD_ROW,
  SET_QUESTION_ALIGNMENT_REMOVE_ROW
} from "../../constants/actions";

describe("question", () => {
  const initialState = {
    entity: {
      alignment: 1
    },
    loading: false,
    saving: false,
    error: null,
    saveError: null
  };

  it("should return the initial state", () => {
    expect(questionReducer(initialState, {})).toMatchSnapshot();
  });

  it("should return the receive question request state", () => {
    expect(
      questionReducer(initialState, {
        type: RECEIVE_QUESTION_REQUEST,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the receive question success state", () => {
    expect(
      questionReducer(initialState, {
        type: RECEIVE_QUESTION_SUCCESS,
        payload: {
          entity: {
            data: {
              type: "multipleChoice",
              stimulus: "<p>What is the capital city of England?</p>",
              ui_style: {
                type: "horizontal"
              },
              options: [
                {
                  value: 0,
                  label: "Red"
                },
                {
                  value: 1,
                  label: "Violet"
                },
                {
                  value: 2,
                  label: "Green"
                }
              ],
              validation: {
                scoring_type: "exactMatch",
                valid_response: {
                  score: 1,
                  value: [1]
                },
                alt_responses: []
              },
              multiple_responses: false,
              smallSize: true
            }
          }
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the receive question error state", () => {
    expect(
      questionReducer(initialState, {
        type: RECEIVE_QUESTION_ERROR,
        payload: {
          error: "unexpected error happened when receive question"
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the save question request state", () => {
    expect(
      questionReducer(initialState, {
        type: SAVE_QUESTION_REQUEST,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the save question success state", () => {
    expect(
      questionReducer(initialState, {
        type: SAVE_QUESTION_SUCCESS,
        payload: {}
      })
    ).toMatchSnapshot();
  });

  it("should return the save question error state", () => {
    expect(
      questionReducer(initialState, {
        type: SAVE_QUESTION_ERROR,
        payload: { error: "unexpected error happend when save question" }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question data state", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION_DATA,
        payload: {
          data: {
            type: "multipleChoice",
            stimulus: "<p>What is the capital city of England?</p>",
            ui_style: {
              type: "horizontal"
            },
            options: [
              {
                value: 0,
                label: "Red"
              },
              {
                value: 1,
                label: "Violet"
              },
              {
                value: 2,
                label: "Green"
              }
            ],
            validation: {
              scoring_type: "exactMatch",
              valid_response: {
                score: 1,
                value: [1]
              },
              alt_responses: []
            },
            multiple_responses: false,
            smallSize: true
          }
        }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question  state", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION,
        payload: { data: {} }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question  state", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION,
        payload: { data: {} }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question alignment add row state1", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION_ALIGNMENT_ADD_ROW,
        payload: { alignmentRow: 1 }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question alignment add row state2", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION_ALIGNMENT_ADD_ROW,
        payload: { alignmentRow: 3 }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question alignment remove row state1", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION_ALIGNMENT_REMOVE_ROW,
        payload: { index: 1 }
      })
    ).toMatchSnapshot();
  });

  it("should return the set question alignment remove row state1", () => {
    expect(
      questionReducer(initialState, {
        type: SET_QUESTION_ALIGNMENT_REMOVE_ROW,
        payload: { index: 3 }
      })
    ).toMatchSnapshot();
  });
});
