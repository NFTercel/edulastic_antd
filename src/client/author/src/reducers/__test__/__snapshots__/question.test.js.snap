// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`question should return the initial state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the receive question error state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": "unexpected error happened when receive question",
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the receive question request state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": null,
  "loading": true,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the receive question success state 1`] = `
Object {
  "entity": Object {
    "data": Object {
      "multiple_responses": false,
      "options": Array [
        Object {
          "label": "Red",
          "value": 0,
        },
        Object {
          "label": "Violet",
          "value": 1,
        },
        Object {
          "label": "Green",
          "value": 2,
        },
      ],
      "smallSize": true,
      "stimulus": "<p>What is the capital city of England?</p>",
      "type": "multipleChoice",
      "ui_style": Object {
        "type": "horizontal",
      },
      "validation": Object {
        "alt_responses": Array [],
        "scoring_type": "exactMatch",
        "valid_response": Object {
          "score": 1,
          "value": Array [
            1,
          ],
        },
      },
    },
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the save question error state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": null,
  "loading": false,
  "saveError": "unexpected error happend when save question",
  "saving": false,
}
`;

exports[`question should return the save question request state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": true,
}
`;

exports[`question should return the save question success state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the set question  state 1`] = `
Object {
  "entity": Object {
    "data": Object {},
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the set question  state 2`] = `
Object {
  "entity": Object {
    "data": Object {},
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;

exports[`question should return the set question data state 1`] = `
Object {
  "entity": Object {
    "alignment": 1,
    "data": Object {
      "multiple_responses": false,
      "options": Array [
        Object {
          "label": "Red",
          "value": 0,
        },
        Object {
          "label": "Violet",
          "value": 1,
        },
        Object {
          "label": "Green",
          "value": 2,
        },
      ],
      "smallSize": true,
      "stimulus": "<p>What is the capital city of England?</p>",
      "type": "multipleChoice",
      "ui_style": Object {
        "type": "horizontal",
      },
      "validation": Object {
        "alt_responses": Array [],
        "scoring_type": "exactMatch",
        "valid_response": Object {
          "score": 1,
          "value": Array [
            1,
          ],
        },
      },
    },
  },
  "error": null,
  "loading": false,
  "saveError": null,
  "saving": false,
}
`;
